lib/dynamic-cache.js

```diff
@@ -35,15 +37,31 @@ DynamicCache.prototype.update = function (cb) {
       this._queue = [];
     }
   }.bind(this);
+
   var getDeps = function (filename) {
-    if (this._files[filename]) {
-      return [[filename]].concat(this._files[filename].map(getDeps)).reduce(function (a, b) {
```

Comment:
The bug in this is actually just that it double counts things, leading to circular structures creating an infinite loop.